stardoc/private/stardoc.bzl

```diff
@@ -0,0 +1,188 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Starlark rule for stardoc: a documentation generator tool written in Java."""
+
+load("@bazel_skylib//:bzl_library.bzl", "StarlarkLibraryInfo")
+
+def _stardoc_runfiles_impl(ctx):
+    runfiles = ctx.runfiles(
+        files = [ctx.file.input],
+        transitive_files = depset(transitive = [
+            dep[StarlarkLibraryInfo].transitive_srcs
+            for dep in ctx.attr.deps
```

Comment:
Looking at the implementation of `bzl_library`, it seems it is already returning transitive files within its runfiles. It would be nicer if you could just drop this additional rule and target and add input,deps directly stardoc_runfiles target. Is there anything preventing that?

Reply:
Also removed the target, please take a look as I may have missed an edge case in the `data = [input] + deps` logic.